src/internal/connector/onedrive/mock/handlers.go

```diff
@@ -0,0 +1,125 @@
+package mock
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+// ---------------------------------------------------------------------------
+// Backup Handler
+// ---------------------------------------------------------------------------
+
+type BackupHandler struct {
```

Comment:
We have an interface with the same name. Looks like this struct is implementing that interface. 
Can this be named `MockBackupHandler`?

Reply:
That would stutter, since the package name is also mock.  IE: `mock.MockFoo`.  Since the declarations are separated by package, the difference between `onedrive.BackupHandler` and `mock.BackupHandler` helps to produce that clarity.

Per the duplication between a struct and an interface name, I do like that less.  And in production code (ie, non-mock) I'm a little more critical about these overlaps.  But here, I think, it works out to read like "mock.BackupHandler means it mocks the BackupHandler interface".